include/deal.II/base/exceptions.h

```diff
@@ -224,8 +224,8 @@ class ExceptionBase : public std::exception
     class Exception1 : public dealii::ExceptionBase      \
     {                                                    \
     public:                                              \
-      Exception1(const type1 a1) : arg1(a1)              \
-      {} /*NOLINT*/                                      \
+      Exception1(type1 const &a1) : arg1(a1)             \
```

Comment:
Would you be willing to switch this back to `const type1 &`, since that's where we put the `const` in all other cases?

Reply:
No, this doesn't work for pointers:
```
../include/deal.II/base/patterns.h:1399:20: note: candidate constructor not viable: no known conversion from 'const dealii::Patterns::Tuple *' to 'const Patterns::PatternBase *&' for 2nd argument
```

Reply:
Oh, I see. This is because of macro substitution -- this wouldn't happen if `type1` were a template argument. How nefarious. I know how to work around this, but this can be done in a later patch.